plugins/woocommerce-admin/client/marketplace/components/notice/notice.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { useState } from '@wordpress/element';
+import { Icon, check, closeSmall, info, percent } from '@wordpress/icons'; // See: https://wordpress.github.io/gutenberg/?path=/docs/icons-icon--docs
+
+/**
+ * Internal dependencies
+ */
+import './notice.scss';
+import sanitizeHTML from '../../../lib/sanitize-html';
+
+export interface NoticeProps {
+	id: string;
+	children?: JSX.Element;
+	description: string;
+	icon?: string;
+	isDismissible: boolean;
+	variant: string;
+}
+
+type IconKey = keyof typeof iconMap;
+
+// Map the icon name (string) to the actual icon component
+const iconMap = {
+	info,
+	check,
+	percent,
+};
+
+export default function Notice( props: NoticeProps ): JSX.Element {
+	const {
+		id,
+		description,
+		children,
+		icon,
+		isDismissible = true,
+		variant = 'info',
+	} = props;
+	const [ isVisible, setIsVisible ] = useState(
+		localStorage.getItem( `wc-marketplaceNoticeClosed-${ id }` ) !== 'true'
+	);
+
+	const handleClose = () => {
+		setIsVisible( false );
+		localStorage.setItem( `wc-marketplaceNoticeClosed-${ id }`, 'true' );
+	};
+
+	if ( ! isVisible ) return <></>;
+
+	const classes = classNames( 'woocommerce-marketplace__notice', {
+		'woocommerce-marketplace__notice-warning': variant === 'warning',
+		'woocommerce-marketplace__notice-info':
+			variant === 'info' || variant === 'info' || variant === 'error',
+		'woocommerce-marketplace__notice-error': variant === 'error',
+		'woocommerce-marketplace__notice-success': variant === 'success',
```

Comment:
I wonder if all those optional clas names could be replaced with something like: `woocommerce-marketplace__notice-${ variant }`?